packages/runtime/src/useLiveResult.ts

```diff
@@ -0,0 +1,70 @@
+import { Dependency, GlobalOptions } from "./utils";
+import { useSSRCache } from "./SSRCacheProvider";
+import { useEffect, useMemo, useRef, useState } from "react";
+
+export type Subscription = {
+  unsubscribe: () => void;
+}
+
+export type Observable<T> = {
+  subscribe: ({
+    next,
+    error,
+  }: {
+    next: (value: T) => void;
+    error?: (error: any) => void;
+  }) => Subscription;
+}
+
+const isObservable = <T>(obj: any): obj is Observable<T> => {
+  return obj && typeof obj.subscribe === 'function';
+}
+
+export const useLiveResult = <ResultT>(
+  dependencies: Dependency[],
+  fn: ((globalOptions: GlobalOptions, ...dependencies: Dependency[])=>any) | undefined,
+  { cachedValue, cacheKey, skip }: { cachedValue: ResultT | null; cacheKey: string; skip: boolean },
+) => {
+  const { globalOptions } = useSSRCache();
+  const observerRef = useRef<Subscription | null>(null);
+  const [subscriptionResult, setSubscriptionResult] = useState<ResultT | null>(null);
+
+  const result = useMemo(() => {
+    observerRef.current?.unsubscribe();
+    observerRef.current = null;
+    if (skip) {
+      return null;
+    }
+
+    if (!fn) {
+      return cachedValue || null;
+    }
+
+    return fn(globalOptions, ...dependencies);
```

Comment:
We should call the function to get the `subscribe` method in the current implementation. However, `fn` can be computationally heavy, resulting in an extra call of `fn`. Seems like a performance issue, right?

If you agree, can we have a separate export of the `subscribe` function in `ssr-computation.js` files?

An ssr-computation will be considered subscribable if there is a `subscribe` function exported from the module.